sw/airborne/modules/system_identification/pprz_chirp.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) Joost Meulenbeld
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file "modules/system_identification/pprz_chirp.c"
+ * @author Joost Meulenbeld
+ * Mathematical implementation of the chirp
+ */
+#include "pprz_chirp.h"
+
+static inline float min(float a, float b) {
+  return a < b ? a : b;
+}
+
+void chirp_init(struct chirp_t* chirp, float f0_hz, float f1_hz, float length_s, float current_time_s, bool exponential_chirp, bool fade_in) {
+  chirp->f0_hz = f0_hz;
+  chirp->f1_hz = f1_hz;
+
+  chirp->length_s = length_s;
+  if (fade_in) // The fade-in takes two of the longest wave-lengths, total_length is including that time
+    chirp->total_length_s = length_s + 2 / f0_hz;
+  else
+    chirp->total_length_s = length_s;
+
+  chirp->start_time_s = current_time_s;
```

Comment:
It's mostly there for debugging purposes so in the message you can see how long it has been since the start of the chirp (via percentage done), and how long still to go.
I will update current_time_s in this place as well.

Reply:
... Not really debugging but progress monitoring